src/kfactory/kcell.py

```diff
@@ -1484,6 +1484,317 @@ class Constants(BaseSettings):
+    info: Info = Info()
+
+
+class LayerStack(BaseModel):
+    """For simulation and 3D rendering.
+
+    Parameters:
+        layers: dict of layer_levels.
+    """
+
+    layers: dict[str, LayerLevel] = Field(default_factory=dict)
+
+    def __init__(self, **data: Any):
+        """Add LayerLevels automatically for subclassed LayerStacks."""
+        super().__init__(**data)
+
+        for field, val in data.items():
+            if not isinstance(val, LayerLevel):
+                raise TypeError(
+                    f"argument should be of type LayerLevel, got type {type(val)}."
+                )
+            self.layers[field] = val
+            if isinstance(val.layer, LayerEnum):
+                self.layers[field].layer = (val.layer[0], val.layer[1])
+
+    def get_layer_to_thickness(self) -> dict[tuple[int, int] | LayerEnum, int]:
+        """Returns layer tuple to thickness (um)."""
+        return {
+            level.layer: level.thickness
+            for level in self.layers.values()
+            if level.thickness
+        }
+
+    def get_layer_to_zmin(self) -> dict[tuple[int, int] | LayerEnum, int]:
+        """Returns layer tuple to z min position (um)."""
+        return {
+            level.layer: level.zmin for level in self.layers.values() if level.thickness
+        }
+
+    def get_layer_to_material(self) -> dict[tuple[int, int] | LayerEnum, str]:
+        """Returns layer tuple to material name."""
+        return {
+            level.layer: level.material
+            for level in self.layers.values()
+            if level.thickness and level.material
+        }
+
+    def get_layer_to_sidewall_angle(self) -> dict[tuple[int, int] | LayerEnum, int]:
+        """Returns layer tuple to material name."""
+        return {
+            level.layer: level.sidewall_angle
+            for level in self.layers.values()
+            if level.thickness
+        }
+
+    def get_layer_to_info(self) -> dict[tuple[int, int] | LayerEnum, Info]:
+        """Returns layer tuple to info dict."""
+        return {level.layer: level.info for level in self.layers.values()}
+
+    def to_dict(self) -> dict[str, dict[str, Any]]:
+        return {level_name: dict(level) for level_name, level in self.layers.items()}
+
+    def __getitem__(self, key: str) -> LayerLevel:
+        """Access layer stack elements."""
+        if key not in self.layers:
+            layers = list(self.layers.keys())
+            raise ValueError(f"{key!r} not in {layers}")
+
+        return self.layers[key]
+
+
+# def get_layer_stack(
+thickness_wg = 220
+thickness_slab_deep_etch = 90
+thickness_clad = 3.0 / 0.001
+thickness_nitride = 350
+thickness_ge = 500
+gap_silicon_to_nitride = 100
+zmin_heater = 1.1 / 0.001
+zmin_metal1 = 1.1 / 0.001
+thickness_metal1 = 700
+zmin_metal2 = 2.3 / 0.001
+thickness_metal2 = 700
+zmin_metal3 = 3.2 / 0.001
+thickness_metal3 = 2000
+substrate_thickness = 10.0 / 0.001
+box_thickness = 3.0 / 0.001
+undercut_thickness = 5.0 / 0.001
```

Comment:
Why is there stuff defined outside the class? And can we remove all the commented stuff below?